File: src/main/java/io/zeitwert/ddd/property/model/base/PropertyFilter.java
package io.zeitwert.ddd.property.model.base;

import javassist.util.proxy.MethodFilter;

import java.lang.reflect.Method;
import java.lang.reflect.Modifier;

public class PropertyFilter implements MethodFilter {

	public static final PropertyFilter INSTANCE = new PropertyFilter();

	@Override
	public boolean isHandled(Method method) {
		return Modifier.isAbstract(method.getModifiers());
	}

}

File: fm-dddrive/src/main/java/io/dddrive/property/model/impl/PropertyFilter.kt
package io.dddrive.property.model.impl

import javassist.util.proxy.MethodFilter
import java.lang.reflect.Method
import java.lang.reflect.Modifier

class PropertyFilter : MethodFilter {

	override fun isHandled(method: Method): Boolean = Modifier.isAbstract(method.modifiers)

	companion object {

		@JvmField
		val INSTANCE: PropertyFilter = PropertyFilter()
	}

}
